### PR TITLE
Create documentation for versioning

### PR DESCRIPTION
**Pull request by MarcinAman**

### CODE REVIEW

**Comment by MarcinAman** [comment]
@kamildoleglo should we state that for now it is only available for mutlimodule or just try to make it happen before the release ?

File: docs/src/doc/docs/user_guide/versioning/versioning.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Versioning plugin
+
+Versioning plugin aims to provide users with ability to create a versioned documentation.
+Therefore, users of the documentation can view different versions of the documentation by going to the main page 
+(for a single module projects this is module page, and for multimodule ones it is the all modules page) and change versions.
+
+Versioning plugin is enabled by default in dokka so there is no need to apply it manually.
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/versioning/versioning.md:7]: dokka -> Dokka

Maybe "is applied by default but not enabled" or similar?

File: docs/src/doc/docs/user_guide/versioning/versioning.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Versioning can be configured using:
+
+* version - a string value representing a version that should be displayed in the dropdown.
+* olderVersionsDir - an optional file that represents the parent directory containing folders with previous dokka outputs.
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/versioning/versioning.md:12]: dokka -> Dokka

File: docs/src/doc/docs/user_guide/versioning/versioning.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+* versionsOrdering - an optional list of strings representing the ordering of versions that should be visible. 
+  By default, dokka will try to use semantic versioning to create such ordering.
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/versioning/versioning.md:14]: dokka -> Dokka

File: docs/src/doc/docs/user_guide/versioning/versioning.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+.
+└── older_versions_dir
+    └── 1.4.10
+        ├── <dokka output>
+    └── 1.4.20
+        ├── <dokka output>
+    ...
+```
+
+As can be seen on the diagram, `olderVersionsDir` should be a parent directory of previous output directories.
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/versioning/versioning.md:33]: I'd also include a example configuration eg. for Gradle

**Review by kamildoleglo — CHANGES_REQUESTED** [review]

**Comment by kamildoleglo** [comment]
I'd state that it will only work for multimodule for now and fix it in the next release

**Pull request merged**